src/emulator.cpp

```diff
@@ -360,7 +360,7 @@ bool Emulator::cbEnvironment(unsigned cmd, void* data) {
 		return false;
 	}
 	case RETRO_ENVIRONMENT_GET_SYSTEM_DIRECTORY:
-		*reinterpret_cast<const char**>(data) = corePath().c_str();
+		*reinterpret_cast<const char**>(data) = strdup(corePath().c_str());
```

Comment:
This seems leaky. Also, should probably be in a separate PR.

Reply:
I only added this because I used valgrind and saw that you use this after it's deallocated which means probably outside the scope of this function.

Reply:
While this is true that means it leaks memory as is and should probably be cached somewhere.

Reply:
Pulled out in #156 so we can continue the discussion there. Not sure who is supposed to own the memory pointed to by `data` and free it eventually, but the way this is currently implemented does not give me any valgrind errors whereas skipping the `strdup` I get multiple errors because it's used after being deallocated.